### PR TITLE
feat: Add has_one relations inference so you don't need to pass accounts that are referenced by a has_one

### PR DESCRIPTION
**Pull request by ChewingGlass**
This is similar to PDA inference. In typescript, you can now avoid passing to `.accounts()` accounts that can be resolved by fetching and pulling fields off of provided public keys. 

Also added a few handy methods I'd like to see. Can remove if needed. 

See the tests/ repo for more details.

### CODE REVIEW

**Comment by staccDOTsol** [comment]
Nooahhhhh

File: lang/syn/src/idl/file.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ pub fn parse(
     filename: impl AsRef<Path>,
     version: String,
     seeds_feature: bool,
+    relations_feature: bool,
```
Comment (armaniferrante) [inline lang/syn/src/idl/file.rs:21]: Should we combine this with the seeds_feature ?

Comment (armaniferrante) [reply lang/syn/src/idl/file.rs:21]: The nomenclature isn't accurate but the intention is the same.

**Review by armaniferrante — COMMENTED** [review]
Please remove the package-lock.json file and the yarn error log.

File: lang/syn/src/idl/file.rs (hunk 1 above)
Comment (ChewingGlass) [reply lang/syn/src/idl/file.rs:21]: Up to you. You mean keep the same name? `seeds = true`

Comment (armaniferrante) [reply lang/syn/src/idl/file.rs:21]: Yes.

**Pull request merged**